joomla/joomla-cms
# [4.0] Separate user/template provided overrides

Pull Request for Issue # .

### The problem

Let's do a simple case study:
Imagine that you're a dev/designer/admin/etc and you decided to use a 3rd PD template. To make this more realistic let's say that you want to use @C-Lodder https://github.com/C-Lodder/joomla4-backend-template 
Few days later you decide that some changes in one of the overrides will fit better your use case. You ego ahead and do the changes. Couple days later Joomla says that the template has an update and you go ahead and apply the update. 

**Ooops your overrides have been overridden**

Ok, apart from this case also the core itself provides overrides https://github.com/joomla/joomla-cms/tree/4.0-dev/administrator/templates/atum/html which **CANNOT** be overridden by the user. So, to cut the story sort here, there are 2 **MAJOR** problems here:
- User is not able to actually override everything (as the project is advertising)
- There is no way any template to provide safe, self owned overrides and also allow the user to use/create their own ones

## The solution
Provide another `html`, I've named it `base_html`, lookalike folder in the template. 

### Summary of Changes
First and foremost the changes here are totally B/C.
The only changes for this to work are:

- add the `base_html` based files as a lookup (2nd in order, html is always the winner if exists) in the MVC, modules and JLayouts

- Patch the override creation process so if a template provided override exists then this will be copied over to the HTML folder


### Testing Instructions
Create some overrides, component, module, layout (use the Atum as your base as it's easier).
Edit each of the overridden file and add some simple text, so you can understand which file was used for the rendering of the page
Rename the html folder to base_html. You should be able to see the same texts
Create the same overrides you did before
The files inside the html folder should have the texts from the base_html files. Rename these texts. Now the rendering should display the last changes.

@ciar4n as a template provider can you give us some feedback here?

### Documentation Changes Required

Yes, I guess
